lib/model.js

```diff
@@ -2102,10 +2102,12 @@ class Model {
 
         // set createdAt/updatedAt attributes
         if (createdAtAttr && !values[createdAtAttr]) {
-          values[createdAtAttr] = now;
+          let createdAtField = instance.rawAttributes[createdAtAttr].field;
+          values[createdAtField || createdAtAttr] = now;
         }
         if (updatedAtAttr && !values[updatedAtAttr]) {
-          values[updatedAtAttr] = now;
+          let updatedAtField = instance.rawAttributes[updatedAtAttr].field;
+          values[updatedAtField || updatedAtAttr] = now;
         }
```

Comment:
I can move the `let` section but `if` check and `values` assignment should remain below the call. Will that be good ?